ci/script_linux.sh

```diff
@@ -1,7 +1,7 @@
 make
 if [ $COVERALLS = ON ]; then make coveralls; fi
 sudo ldconfig --verbose # So the test executeables can detect libtinyxml2
-if [ $BUILD_CORE_ONLY = OFF ]; then make test; fi
+if [ $BUILD_CORE_ONLY = OFF ]; then make all test tests; fi
```

Comment:
You're right about the order: it should be `make all test tests`. This was a typo.

All does need to be here. `make` will invoke the first target in the `Makefile`, which is typically all, if no goals are specified on the command line:

> By default, the goal is the first target in the makefile (not counting targets that start with a period). Therefore, makefiles are usually written so that the first target is for compiling the entire program or programs they describe. If the first rule in the makefile has several targets, only the first target in the rule becomes the default goal, not the whole list. You can manage the selection of the default goal from within your makefile using the .DEFAULT_GOAL variable (see Other Special Variables). ([Source](http://www.gnu.org/software/make/manual/make.html#Goals))

Since we *are* specifying a goal on the command line, we need to manually pass `all`. In practice, we may not need to do this because the `tests` target may indirectly depend on all of the targets included in `all`.

This raises another important question: should we also build `tutorials` and `examples` here, to verify that they build successfully?